src/app/modules/status.rs

```diff
@@ -353,7 +353,7 @@ pub(crate) struct StatusModuleVTable {
     /// for a status kind that is greater than the current length will return an invalid status kind
     /// * All previous status kinds are cleared by [`StatusModule::change_status_force_and_clear`]
     #[offset = 0x148]
-    pub prev_status_kind: extern "C" fn(this: &StatusModule, index: i32) -> i32,
+    pub prev_status_kind: extern "C" fn(this: &StatusModule, index: u32) -> i32,
```

Comment:
Personally would prefer this to stay as i32 if that is what the symbol `StatusModule::prev_status_kind` takes as it's argument. You can check this in ghidra or with a name demangler. Although the index may in practice be unsigned, it's not my fault if the smash devs used `int` instead of `unsigned int`

Reply:
The symbol for `StatusModule::prev_status_kind` is as follows:
```_ZN3app8lua_bind35StatusModule__prev_status_kind_implEPNS_26BattleObjectModuleAccessorEj```
I believe the `j` indicates an unsigned integer as opposed to a signed integer with `i` instead, hence the change. If I am wrong on this, I will revert the change.